docs/modules/ROOT/pages/how-tos/worklows/git-submodules.adoc

```diff
@@ -0,0 +1,90 @@
+= Git Submodule pattern
+
+== How this workflow pattern works:
+
+1. Create a "downstream" repo.
+2. Track your upstream git repo with a git submodule reference.
+3. Onboard to {ProductName}, use link:https://github.com/renovatebot/renovate[Renovate] to keep your Git submodule references up to date.
+4. In your downstream repo, apply patches in your Containerfile if you need to carry downstream patches.
+
+== Implementing this solution pattern
+=== Create a downstream repo
+
+A **downstream repo** is a self-owned copy of the upstream repository whose build, test, release lifecycle you want to control completely.
+
+The downstream repo will:
+
+* be used to hold the git submodule reference to the upstream repository.
+* contain a Containerfile that references and pulls in the git submodule(s) content.
+* become a Component of our {ProductName} application.
+
+=== Create the git submodule for the upstream repo to track it
+
+An **upstream repo** is a community project which you do not have push access to (or which you do not want to push CI changes into).
+
+After creating the downstream repo, add a git submodule for the upstream repo:
+
+[source, bash]
+----
+git submodule add <upstream-url>
+----
+
+Check and configure the git submodule config:
+[source, gitmodules]
+----
+# .gitmodules
+[submodule "<project-name>"]
+  path = <path-in-downstream-repo>
+  url = https://github.com/<namespace>/<project>.git
+  branch = main # Ensure the branch is correct
```

Comment:
Is there a branch added by default? If not, should we direct users to specifically add this?

Reply:
I added a section under this about configuring the branch value. It's not specified by default from my experience.

Reply:
Is it to prevent the wrong branch or more precisely to prevent the default branch from being used?

Reply:
> Is it to prevent the wrong branch or more precisely to prevent the default branch from being used?

Hmm... it could be both? Maybe someone would prefer another branch besides main, for example a branch a named `version-x` for a specific release

From my own experience it looks like the git submodule defaults to the last commit to that repo, most likely main/master.